operator/v1/types_klusterlet.go

```diff
@@ -162,6 +163,51 @@ type RegistrationConfiguration struct {
 	// +optional
 	// +kubebuilder:default:=100
 	KubeAPIBurst int32 `json:"kubeAPIBurst,omitempty"`
+
+	// PriorityBootstrapKubeconfigs defines the list of bootstrap kubeconfigs in the order of priority.
+	// They are used for an agent to connect to multiple hubs in cases like auto-failover, backup&restore, rollout upgrade etc.
+	//
+	// When the agent loses the connection to the current hub over HubConnectionTimeoutSeconds, or the managedcluster CR
+	// is set `hubAcceptsClient=false` on the hub, we mark the related bootstrap secret as failed.
+	// A failed bootstrapkubeconfig won't be used for the duration specified by SkipFailedBootstrapKubeconfigSeconds.
+	//
+	// Then the agent will try to choose another bootstrapkubeconfig from the list by order, replace the current bootstrapkubeconfig and trigger the rebootstrap.
+	// +optional
+	PriorityBootstrapKubeconfigs *PriorityBootstrapKubeconfigs `json:"priorityBootstrapKubeconfigs,omitempty"`
+
+	// HubConnectionTimeoutSeconds is used to set the timeout of connecting to the hub cluster.
+	// When agent loses the connection to the hub over the timeout seconds, the agent do a rebootstrap.
+	// By default is 10 mins.
+	// +optional
+	// +kubebuilder:default:=600
+	// +kubebuilder:validation:Minimum=0
+	HubConnectionTimeoutSeconds int32 `json:"hubConnectionTimeoutSeconds,omitempty"`
+
+	// SkipFailedBootstrapKubeconfigSeconds. The controller will skip the bootstrapkubeconfigs that fails recently.
```

Comment:
tit is not quite clear what this means. What if there is only one bootstrap-kube-secret, what if secret is updated to the correct value?

Reply:
If there is only one item in the list and it's marked as "failed", the controller will skip it when choose a new bootstrapkubeconfig.

If the secret is updated, the failed mark will be cleared. I also add this in the description of PriorityBootstrapKubeconfigs.

Since this field is highly related to the list, I moved it inside the PriorityBootstrapKubeconfigs structure.